pensieve/config.py

```diff
@@ -26,30 +26,90 @@
 import jinja2
 import mozanalysis.metrics
 import mozanalysis.metrics.desktop
+import pandas
 
-from pensieve.analysis import AnalysisPeriod
+from . import AnalysisPeriod
+from pensieve.statistics import Statistic, BootstrapMean, StatisticResultCollection
 import pensieve.experimenter
 
 
-# TODO: these should be objects that pair a metric with a default statistical treatment.
+@attr.s(auto_attribs=True)
+class Summary:
+    """Represents a metric with a statistical treatment."""
+
+    metric: mozanalysis.metrics.Metric
+    treatment: Statistic
+
+    def run(self, data: pandas.DataFrame) -> "StatisticResultCollection":
+        """Apply the statistic transformation for data related to the specified metric."""
+        return self.treatment.apply(data, self.metric.name)
+
+
+# metrics and statistics calculated for each experiment
 DEFAULT_METRICS = {
     "desktop": {
-        "daily": [mozanalysis.metrics.desktop.unenroll],
-        "weekly": [
-            mozanalysis.metrics.desktop.active_hours,
-            mozanalysis.metrics.desktop.uri_count,
-            mozanalysis.metrics.desktop.ad_clicks,
-            mozanalysis.metrics.desktop.search_count,
+        AnalysisPeriod.DAY: [
+            Summary(
+                metric=mozanalysis.metrics.desktop.unenroll,
+                treatment=BootstrapMean(num_samples=1000),
+            )
         ],
-        "overall": [
-            mozanalysis.metrics.desktop.active_hours,
-            mozanalysis.metrics.desktop.uri_count,
-            mozanalysis.metrics.desktop.ad_clicks,
-            mozanalysis.metrics.desktop.search_count,
+        AnalysisPeriod.WEEK: [
+            Summary(
+                metric=mozanalysis.metrics.desktop.active_hours,
+                treatment=BootstrapMean(num_samples=1000),
+            ),
+            Summary(
+                metric=mozanalysis.metrics.desktop.uri_count,
+                treatment=BootstrapMean(num_samples=1000),
+            ),
+            Summary(
+                metric=mozanalysis.metrics.desktop.ad_clicks,
+                treatment=BootstrapMean(num_samples=1000),
+            ),
+            Summary(
+                metric=mozanalysis.metrics.desktop.search_count,
+                treatment=BootstrapMean(num_samples=1000),
+            ),
+        ],
+        AnalysisPeriod.OVERALL: [
+            Summary(
+                metric=mozanalysis.metrics.desktop.active_hours,
+                treatment=BootstrapMean(num_samples=1000),
+            ),
+            Summary(
+                metric=mozanalysis.metrics.desktop.uri_count,
+                treatment=BootstrapMean(num_samples=1000),
+            ),
+            Summary(
+                metric=mozanalysis.metrics.desktop.ad_clicks,
+                treatment=BootstrapMean(num_samples=1000),
+            ),
+            Summary(
+                metric=mozanalysis.metrics.desktop.search_count,
+                treatment=BootstrapMean(num_samples=1000),
+            ),
```

Comment:
We could also move those into a `.toml` config file that lives in the pensieve repository instead of hardcode them. It might be easier to find and change if there is a separate file for this and could also serve as a configuration example.